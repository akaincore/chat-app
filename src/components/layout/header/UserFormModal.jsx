import React from 'react';
import TextField from '@material-ui/core/TextField';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import Button from '@material-ui/core/Button';

const UserFormModal = ({
                    modalOpen,
                    handleModalClose,
                    userForm,
                    onUserFieldChange,
                    handleUpdateUser
                  }) => {
  const { username, firstName, lastName } = userForm;
  return (
    <Dialog
      open={modalOpen}
      onClose={handleModalClose}
      aria-labelledby="form-dialog-title"
    >
      <DialogTitle id="form-dialog-title">Edit profile</DialogTitle>
      <DialogContent>
        <TextField
          margin="dense"
          name="username"
          label="Username"
          type="text"
          required
          fullWidth
          value={username}
          onChange={onUserFieldChange}
        />
        <TextField
          margin="dense"
          name="firstName"
          label="First name"
          type="text"
          fullWidth
          value={firstName}
          onChange={onUserFieldChange}
        />
        <TextField
          margin="dense"
          name="lastName"
          label="Last name"
          type="text"
          fullWidth
          value={lastName}
          onChange={onUserFieldChange}
        />
      </DialogContent>
      <DialogActions>
        <Button onClick={handleUpdateUser} color="primary">
          Save
        </Button>
        <Button onClick={handleModalClose} color="primary">
          Cancel
        </Button>
      </DialogActions>
    </Dialog>
  );
};

export default UserFormModal;
