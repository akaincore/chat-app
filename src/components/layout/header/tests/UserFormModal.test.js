/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import UserFormModal from '../UserFormModal';

const mockProps = {
  userForm: { foo: 'bar' },
  onUserFieldChange: jest.fn(),
  handleModalClose: jest.fn(),
  handleUpdateUser: jest.fn(),
  modalOpen: false,
};

describe('<UserFormModal />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<UserFormModal {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<UserFormModal {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
