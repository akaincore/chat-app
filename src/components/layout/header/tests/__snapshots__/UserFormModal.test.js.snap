// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserFormModal /> renders correctly 1`] = `null`;
